base_layer/core/src/proof_of_work/proof_of_work.rs

```diff
@@ -25,13 +25,18 @@ use std::fmt::{Display, Error, Formatter};
 use borsh::{BorshDeserialize, BorshSerialize};
 use bytes::BufMut;
 use serde::{Deserialize, Serialize};
+use tari_common_types::MaxSizeBytes;
 use tari_utilities::hex::Hex;
 
 use crate::proof_of_work::PowAlgorithm;
 
 #[allow(dead_code)]
 pub trait AchievedDifficulty {}
 
+/// This is the maximum size of the proof of work data that can be included in a block header. This is used to prev
+pub const NOT_BEFORE_PROOF_BYTES_SIZE: usize = u16::MAX as usize;
```

Comment:
We need to ensure that the monero header + coinbase data can fit in here

Reply:
Agreed. This is quite large at the moment, `assert_eq!(u16::MAX, 65535);`, much larger than a monero header and coinbase data.

https://github.com/monero-project/monero/blob/b089f9ee69924882c5d14dd1a6991deb05d9d1cd/src/cryptonote_basic/cryptonote_basic.h#L457

```c++
  /************************************************************************/
  /*                                                                      */
  /************************************************************************/
  struct block_header
  {
    uint8_t major_version;
    uint8_t minor_version;  // now used as a voting mechanism, rather than how this particular block is built
    uint64_t timestamp;
    crypto::hash  prev_id;
    uint32_t nonce;

    BEGIN_SERIALIZE()
      VARINT_FIELD(major_version)
      VARINT_FIELD(minor_version)
      VARINT_FIELD(timestamp)
      FIELD(prev_id)
      FIELD(nonce)
    END_SERIALIZE()
  };
```